Missing NumberFormatUnit symbol in lib.es2020.intl.d.ts

# Bug Report

<!--
  Please fill in each section completely. Thank you!
-->

### 🔎 Search Terms

- lib
- d.ts

### 🕗 Version & Regression Information

4.5.0-dev.20210909

### 💻 Code

1. Update the VS Code code base to build with the latest TS nightly
2. Run `yarn monaco-compile-check`


### Problem
https://github.com/microsoft/vscode/runs/3558995204?check_suite_focus=true

We see the errors:

```
Error: node_modules/typescript/lib/lib.es2020.intl.d.ts(222,9): error TS2717: Subsequent property declarations must have the same type.  Property 'type' must be of type 'ES2018NumberFormatPartType', but here has type 'ES2020NumberFormatPartType'.
9
Error: node_modules/typescript/lib/lib.es2020.intl.d.ts(229,16): error TS2304: Cannot find name 'NumberFormatUnit'.
10
Error: node_modules/typescript/lib/lib.es2020.intl.d.ts(237,16): error TS2304: Cannot find name 'NumberFormatUnit'.
```

I searched through the libs but also couldn't find where `NumberFormatUnit` is defined. Let me know if I've overlooked something silly

/cc @orta

--- a/src/lib/es2018.intl.d.ts
+++ b/src/lib/es2018.intl.d.ts
@@ -37,9 +37,12 @@ declare namespace Intl {
         supportedLocalesOf(locales: string | string[], options?: { localeMatcher?: "lookup" | "best fit" }): string[];
     };
 
+    // We can only have one definition for 'type' in TypeScript, and so you can learn where the keys come from here:
     type ES2018NumberFormatPartType = "literal" | "nan" | "infinity" | "percent" | "integer" | "group" | "decimal" | "fraction" | "plusSign" | "minusSign" | "percentSign" | "currency" | "code" | "symbol" | "name";
+    type ES2020NumberFormatPartType = "compact" | "exponentInteger" | "exponentMinusSign" | "exponentSeparator" | "unit" | "unknown";
+
     interface NumberFormatPart {
-        type: ES2018NumberFormatPartType;
+        type: ES2018NumberFormatPartType | ES2020NumberFormatPartType;
         value: string;
     }
 

--- a/src/lib/es2020.intl.d.ts
+++ b/src/lib/es2020.intl.d.ts
@@ -197,11 +197,6 @@ declare namespace Intl {
         ): UnicodeBCP47LocaleIdentifier[];
     };
 
-    interface NumberFormatPart {
-        // es2020 ones start after "name"
-        type: "literal" | "nan" | "infinity" | "percent" | "integer" | "group" | "decimal" | "fraction" | "plusSign" | "minusSign" | "percentSign" | "currency" | "code" | "symbol" | "name" | "compact" | "exponentInteger" | "exponentMinusSign" | "exponentSeparator" | "unit" | "unknown";
-    }
-
     interface NumberFormatOptions {
         compactDisplay?: "short" | "long" | undefined;
         notation?: "standard" | "scientific" | "engineering" | "compact" | undefined;

--- a/src/lib/es2020.intl.d.ts
+++ b/src/lib/es2020.intl.d.ts
@@ -214,7 +214,7 @@ declare namespace Intl {
         compactDisplay?: "short" | "long";
         notation?: "standard" | "scientific" | "engineering" | "compact";
         signDisplay?: "auto" | "never" | "always";
-        unit?: stringt;
+        unit?: string;
         unitDisplay?: "short" | "long" | "narrow";
     }
 

--- a/src/lib/es2020.intl.d.ts
+++ b/src/lib/es2020.intl.d.ts
@@ -197,16 +197,16 @@ declare namespace Intl {
         ): UnicodeBCP47LocaleIdentifier[];
     };
 
-    type ES2020NumberFormatPartType = ES2018NumberFormatPartType | "compact" | "exponentInteger" | "exponentMinusSign" | "exponentSeparator" | "unit" | "unknown";
     interface NumberFormatPart {
-        type: ES2020NumberFormatPartType
+        // es2020 ones start after "name"
+        type: "literal" | "nan" | "infinity" | "percent" | "integer" | "group" | "decimal" | "fraction" | "plusSign" | "minusSign" | "percentSign" | "currency" | "code" | "symbol" | "name" | "compact" | "exponentInteger" | "exponentMinusSign" | "exponentSeparator" | "unit" | "unknown";
     }
 
     interface NumberFormatOptions {
         compactDisplay?: "short" | "long" | undefined;
         notation?: "standard" | "scientific" | "engineering" | "compact" | undefined;
         signDisplay?: "auto" | "never" | "always" | undefined;
-        unit?: NumberFormatUnit | undefined;
+        unit?: string | undefined;
         unitDisplay?: "short" | "long" | "narrow" | undefined;
     }
 
@@ -214,7 +214,7 @@ declare namespace Intl {
         compactDisplay?: "short" | "long";
         notation?: "standard" | "scientific" | "engineering" | "compact";
         signDisplay?: "auto" | "never" | "always";
-        unit?: NumberFormatUnit;
+        unit?: stringt;
         unitDisplay?: "short" | "long" | "narrow";
     }
 

--- a/tests/baselines/reference/libCompileChecks.js
+++ b/tests/baselines/reference/libCompileChecks.js
@@ -1,0 +1,7 @@
+//// [libCompileChecks.ts]
+// This test is effectively the 'lib check' for all our .d.ts files because we use skipLibCheck
+// as false everywhere.
+
+//// [libCompileChecks.js]
+// This test is effectively the 'lib check' for all our .d.ts files because we use skipLibCheck
+// as false everywhere.

--- a/tests/baselines/reference/libCompileChecks.symbols
+++ b/tests/baselines/reference/libCompileChecks.symbols
@@ -1,0 +1,4 @@
+=== tests/cases/compiler/libCompileChecks.ts ===
+// This test is effectively the 'lib check' for all our .d.ts files because we use skipLibCheck
+No type information for this code.// as false everywhere.
+No type information for this code.

--- a/tests/baselines/reference/libCompileChecks.types
+++ b/tests/baselines/reference/libCompileChecks.types
@@ -1,0 +1,4 @@
+=== tests/cases/compiler/libCompileChecks.ts ===
+// This test is effectively the 'lib check' for all our .d.ts files because we use skipLibCheck
+No type information for this code.// as false everywhere.
+No type information for this code.

--- a/tests/cases/compiler/libCompileChecks.ts
+++ b/tests/cases/compiler/libCompileChecks.ts
@@ -1,0 +1,7 @@
+// @lib: esnext, dom
+// @target: esnext
+// @skipLibCheck: false
+// @skipDefaultLibCheck: false
+
+// This test is effectively the 'lib check' for all our .d.ts files because we use skipLibCheck
+// as false everywhere.

